OMGVA PoS/Business layer/Controllers/TaxController.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using OMGVA_PoS.Data_layer.DTOs;
+using OMGVA_PoS.Data_layer.Mappers;
+using OMGVA_PoS.Data_layer.Repositories.Tax;
+
+namespace OMGVA_PoS.Business_layer.Controllers
+{
+    [ApiController]
+    [Route("tax")]
+    public class TaxController(ITaxRepository taxRepository) : ControllerBase
+    {
+
+        private static string TaxNotFound(long id) => $"Tax with ID {id} not found.";
+
+        [HttpPost]
+        public async Task<IActionResult> CreateTax([FromBody] CreateTaxRequest createTaxRequest)
+        {
+            var tax = TaxMapper.FromCreateTaxRequest(createTaxRequest);
+            await taxRepository.SaveTaxAsync(tax);
+
+            return CreatedAtAction(nameof(GetTaxById), new { tax.Id }, tax);
+        }
+
+        [HttpPut("{id}")]
```

Comment:
In our previous controllers, we added authorization and status code attributes. I would recommend you to do the same. For example, if tax can be updated by admin only, then you should add:
`[Authorize(Roles = "Admin")]`
For status codes to be seen clearer in the swagger:
`[ProducesResponseType<TypeIfNeeded>(StatusCodes.Status400BadRequest)]`